core/mesh/peer.go

```diff
@@ -59,9 +60,19 @@ func (s Machine) ToProtobuf() *pb.Peer {
 		Os:         s.OS.Name,
 		Distro:     s.OS.Distro,
 		Hostname:   s.Hostname,
+		Nickname:   s.Nickname,
 	}
 }
 
+func (s *Machine) IsEqual(b Machine) bool {
+	return s.ID == b.ID &&
+		s.HardwareID == b.HardwareID &&
+		s.Hostname == b.Hostname &&
+		s.Address == b.Address &&
+		s.SupportsRouting == b.SupportsRouting &&
+		s.Nickname == b.Nickname
```

Comment:
Looks like too many conditions: `Nickname` is not needed because of `Hostname`, and why you need to check `SupportsRouting`? it's not part of ID. Or what situation you are trying to handle?

Reply:
Ok, I got it - initially I thought it is to find Machine, but below I see it is to check if there were any changes made to machine data.

Reply:
we store locally the info about the current machine. So I needed to check if 2 objects of this type are equal. Not sure if the other 3 `OS`, `PublicKey` and `Endpoints` makes sense to check too...